components/FilteredTextArea.tsx

```diff
@@ -15,16 +15,41 @@ export default function FilteredTextArea(props: FilteredTextAreaProps) {
 
   const handleChange = useCallback(
     (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      e.currentTarget.value = [
+        ...chunks(
+          [
+            ...chunks(
+              e.currentTarget.value.replace(/[^a-z0-9]/gi, "").split(""),
+              2
+            ),
+          ].map((pair) => pair.join("")),
+          6
+        ),
+      ]
+        .map((row) => row.join(" "))
+        .join("\n");
       const newValue = e.currentTarget.value;
       const validationMsg = validate(newValue);
       textAreaEl.current.setCustomValidity(validationMsg);
 
-      onChange && onChange(e);
+      if (onChange) {
+        onChange(e);
+      }
     },
     [validate, textAreaEl, onChange]
   );
 
   return (
-    <textarea ref={textAreaEl} {...textAreaProps} onChange={handleChange} />
+    <textarea
+      ref={textAreaEl}
+      {...textAreaProps}
+      onChange={handleChange}
+      data-testid="foo"
```

Comment:
Could pass down from parent?

Reply:
Sure—why?

Reply:
I originally thought we could just pass it via the testing library, but now that I think about it, it probably doesn't matter. Maybe could just rename it to `filtered-text-area` or something 😄